lib/services/local/integrations.go

```diff
@@ -100,10 +118,56 @@ func (s *IntegrationsService) UpdateIntegration(ctx context.Context, ig types.In
 
 // DeleteIntegration removes the specified Integration resource.
 func (s *IntegrationsService) DeleteIntegration(ctx context.Context, name string) error {
-	return trace.Wrap(s.svc.DeleteResource(ctx, name))
+	conditionalActions, err := notReferencedByEAS(ctx, s.backend, name)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	conditionalActions = append(conditionalActions, backend.ConditionalAction{
+		Key:       s.svc.MakeKey(name),
+		Condition: backend.Whatever(),
```

Comment:
The preparation phase should check that the integration exists and error out with a `trace.NotFound` if it doesn't, and the write should assert that it still exists.

```suggestion
		Condition: backend.Exists(),
```